site/topics/debug/debug.rst

```diff
@@ -28,60 +28,107 @@ The Importance of Debugging
     * The ability to reason logically
     * The ability to reason about flow of execution in code
 
-   
-Syntax errors
-=============
-* These are easy to fix. 
-* You make a typo and Python tells you the problem, and where it is, straight away.
-* You fix the syntax and... problem solved.
-* Not much strategy: just look at the code or look at what python yells at you about
+
+Error Messages
+==============
+
+* In many cases, Python will generate an error message giving you an idea of what went wrong
+* Sometimes these error messages are very helpful
+
+    .. code-block:: python
+
+        line 17
+        print(5 + 5 +)
+                    ^
+        SyntaxError: invalid syntax
+
+
+* Other times the error messages are quite cryptic
+
+    * Fortunately, you can often copy/paste the error into Google and find an explanation
+
+* Common error messages that you will come across (if you have not already) are
+
+    * Syntax errors
+    * Indentation Errors
+    * Name Errors
+    * Type Error
+    * Index Error
+
+
+Syntax Errors
+-------------
+
+* Syntax errors will cause error messages that are often quite helpful
+* These messages typically tell you where the error is, as seen in the above example
+* There is not much of a debugging strategy with these errors as you just look at what Python tells you
 
 .. admonition:: Activity
     :class: activity
    
-    Fix this Python function::
+    See if you can find all the errors in the following example. You may be able to find them all just by looking at the
+    code, but feel free to copy/paste the code into Colab to see if the error messages help.
+
+    .. code-block:: python
+        :linenos:
    
         deff borken(a(
             a = a + * 3
             walrus
             return a + 2   
 		 
         broken(5)
-   
-Type errors
-===========
 
-* As we've seen many times, Python is pretty good at transparently guessing how to change types when you ask it do something that involves multiple types::
-  
-    >>> 2.03 + 4
-    6.0299999999999994
-     
-* **But** sometimes you might ask the impossible:
 
-     >>> 2.03 + 'octopus'
-     TypeError: unsupported operand type(s) for +: 'float' and 'str'
+Type Errors
```

Comment:
Think you're missing a new line after the code block. Syntax highlighting is outta whack.